compiler/hash-lower/src/discover.rs

```diff
@@ -160,7 +160,6 @@ impl FnDiscoverer<'_> {
         self.traversing_utils()
             .visit_term::<!, _>(term, &mut |atom: Atom| match atom {
                 Atom::Term(_) => Ok(ControlFlow::Continue(())),
-                Atom::Ty(_) => Ok(ControlFlow::Break(())),
```

Comment:
It's not strictly necessary to change this, but should the updated version also break on type-like terms?

Reply:
Technically the correct thing to do is to recurse in case that there are indeed some functions in there. For example `(a: i32, b: i32 -> i32 = (x) => x)` is a valid type.

Reply:
But in that example, it is in the `default` expression position - not inside the type itself.

Reply:
Yes but the root node is a `TupleTy` so if `Break` is given for that then the function will never be seen.

Reply:
I know that technically default values are not supported yet but this is a side point

Reply:
Yeah ok, this makes sense. 

Side note: In the current implementation, this wouldn't quite work anyway (I think).

Reply:
Yeah it wouldn't, but I hope soon it will